eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryImpl.java

```diff
@@ -126,7 +140,9 @@ private void addFilterCriteriaToQuery(Query query, List<VariantEntityRepositoryF
 
     private long countByComplexFiltersHelper(Criteria existingCriteria, List<VariantEntityRepositoryFilter> filters) {
         List<Criteria> criteriaList = getFiltersCriteria(filters);
-        criteriaList.add(existingCriteria);
+        if (existingCriteria != null) {
```

Comment:
is it correct to omit `existingCriteria` if it's null? if it should always be not null, this `if` is bad: it delays the detection of a missing value, and it would be better to add an else with an explanatory message (if the field requires clarification why it should not be null) or just let it throw the NPE.